.github/workflows/iroha2-dev-pr.yml

```diff
@@ -144,44 +150,67 @@ jobs:
           name: clippy.json
           path: clippy.json
 
-  torii-api-and-client-cli-tests:
+  torii-api-client-cli-docker-compose-tests:
     runs-on: [self-hosted, Linux, iroha2]
-    container:
-      image: hyperledger/iroha2-ci:nightly-2024-04-18
     timeout-minutes: 60
     steps:
-      - uses: actions/checkout@v4
-      - uses: Swatinem/rust-cache@v2
-      - name: Build binaries
-        run: |
-          cargo build -p iroha_client_cli -p kagami -p irohad
-      - name: Setup test Iroha 2 environment on the bare metal
-        run: |
-          pip3 install -r scripts/requirements.txt --no-input --break-system-packages
-          ./scripts/test_env.py setup
-      - name: Mark binaries as executable
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Set up Docker Buildx
+        id: buildx
+        if: always()
+        uses: docker/setup-buildx-action@v3
+        with:
+          install: true
+      - name: Install Python and Poetry
         run: |
-          chmod +x ${{ env.CLIENT_CLI_DIR }}
-      - name: Install torii api dependencies using Poetry
+          yum install -y python${{ env.PYTHON_VERSION }} python${{ env.PYTHON_VERSION }}-devel
+          curl -sSL https://install.python-poetry.org | python3 -
+          echo 'export PATH="${{ env.POETRY_PATH }}:$PATH"' >> /etc/profile
+          source /etc/profile
```

Comment:
What image is used here? Can we use image with `python` and `poetry` already installed? not to waste CI time

Reply:
As far as I know it uses Amazon Linux 2023. For some reason we can't pre-install it, hopefully @BAStos525 will let us know.

Reply:
We can't use ci-image here because Actions can not run containers inside another (ci) container properly. It means when we run iroha containers inside ci-container, they just exited. So, we push to run then on the CI runner directly.

Reply:
Why can't we customize the default Amazon Linux 2023 image?

Reply:
Because it is limited by the current technology of using CI spot runners